### PR TITLE
add docs upstream validation workflow

### PR DESCRIPTION
**Pull request by dvdksn**
Signed-off-by: David Karlsson <35727626+dvdksn@users.noreply.github.com>

**What I did**

added a gha workflow that verifies that the reference docs don't contain broken links etc

**Related issue**

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

🐥

### CODE REVIEW

**Comment by dvdksn** [comment]
hey @glours and @ndeloof, I added this new workflow that gives you some validation (check for broken links etc) in your CI. We have the same in Buildx and BuildKit, and since we vendor docs from this repo too I thought it would be good to add here

File: .github/workflows/docs-upstream.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+name: docs-upstream
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches:
+      - 'master'
```
Comment (glours) [inline .github/workflows/docs-upstream.yml:12]: `master`? 🤔 
```suggestion
      - 'main'
```

Comment (dvdksn) [reply .github/workflows/docs-upstream.yml:12]: Oops :)

**Review by glours — APPROVED** [review]
LGTM

**Pull request merged**